www/docs/en/dev/guide/platforms/ios/plugin.md

```diff
@@ -159,84 +205,130 @@ specification to the local platform's `config.xml` file:
-method get the first parameter of the `arguments` array, which
-corresponds to the arguments passed in by the JavaScript `exec()`
-function.
+For example, if the plugin has defined `onload` as `true`, when the app loads, the `pluginInitialize` method will be executed. Because this is triggered during app load, there is no `callbackID` so the `pluginInitialize` method can not return any results to the WebView. If results matter, they would need to be stored in some manar and later fetched with a JavaScript API call.
 
-It checks the parameter to make sure it is not `nil` or an empty
-string, returning a `PluginResult` with an `ERROR` status if so.  If
-the parameter passes the check, it returns a `PluginResult` with an
-`OK` status, passing in the original `echo` string.  Finally, it sends
-the result to `self.commandDelegate`, which executes the `exec`
-method's success or failure callbacks on the JavaScript side. If the
-success callback is called, it passes in the `echo` parameter.
+#### Handeling Long-running & Background Activities
 
-## iOS Integration
+Plugins with long-running requests or background activities, such as media playback, listeners, or those that maintain internal state, should implement the `onReset` method to cancel these requests or clean up after those activities.
 
-The `CDVPlugin` class features other methods that your plugin can
-override.  For example, you can capture the [pause][PauseEvent], [resume][ResumeEvent], app
-terminate and `handleOpenURL` events. See the
-[CDVPlugin.h][CDVPlugin.h] and [CDVPlugin.m][CDVPlugin.m]
-classes for guidance.
+The `onReset` method is called when the `WKWebView` navigates to a new page or refreshes, triggering a reload of the JavaScript.
 
-### WKURLSchemeTask Hook
+#### Hooking into WKURLSchemeTask
 
-The [WKURLSchemeTask](https://developer.apple.com/documentation/webkit/wkurlschemetask) is an interface Cordova's main WKWebView uses to load files from your app's bundle. You can create your own custom schemes or custom loading code for the webview by implementing the `- (BOOL) overrideSchemeTask: (id <WKURLSchemeTask>)urlSchemeTask` method in a plugin.
+The [WKURLSchemeTask](https://developer.apple.com/documentation/webkit/wkurlschemetask) is an interface Cordova's main WKWebView uses to load files from your app's bundle. You can create your own custom schemes or custom loading code for the WebView by implementing the `- (BOOL) overrideSchemeTask: (id <WKURLSchemeTask>)urlSchemeTask` method in a plugin.
 
-## Threading
+#### Using Background Threads
 
-Plugin methods ordinarily execute in the same thread as the main
-interface. If your plugin requires a great deal of processing or
-requires a blocking call, you should use a background thread. For
-example:
+Plugin methods ordinarily execute in the same thread as the main interface. If your plugin requires a great deal of processing or requires a blocking call, you should use a background thread.
```

Comment:
Worth mentioning that certain operations involving UI stuff (showing alerts, changing colours, etc.) can only run on the main thread